### PR TITLE
Artefact Builds for the Language Server

### PR DESCRIPTION
**Pull request by lolczak**
### Pull Request Description
Language Server runner. Artefact build fix.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runner/src/main/scala/org/enso/runner/Main.scala [hunk 1]
```diff
@@ -1,71 +1,112 @@
...
-  private val NEW_OPTION     = "new"
-  private val REPL_OPTION    = "repl"
-  private val JUPYTER_OPTION = "jupyter-kernel"
+  private val RUN_OPTION             = "run"
+  private val HELP_OPTION            = "help"
+  private val NEW_OPTION             = "new"
+  private val REPL_OPTION            = "repl"
+  private val JUPYTER_OPTION         = "jupyter-kernel"
+  private val LANGUAGE_SERVER_OPTION = "server"
+  private val INTERFACE_OPTION       = "i"
```
Comment (kustosz) [inline engine/runner/src/main/scala/org/enso/runner/Main.scala]: please use long options for everything

File: engine/runner/src/main/scala/org/enso/runner/Main.scala [hunk 2]
```diff
@@ -164,6 +205,50 @@ object Main {
...
+    * Handles `--language-server` CLI option
+    *
+    * @param line a CLI line
+    */
+  private def runLanguageServer(line: CommandLine): Unit = {
+    val maybeConfig = parseSeverOptions(line)
+
+    maybeConfig match {
+      case Left(errorMsg) =>
+        System.err.println(errorMsg)
```
Comment (kustosz) [inline engine/runner/src/main/scala/org/enso/runner/Main.scala]: A failure exit should follow?

File: engine/runner/src/main/scala/org/enso/runner/Main.scala [hunk 3]
```diff
@@ -164,6 +205,50 @@ object Main {
...
+        exitSuccess()
+    }
+  }
+
+  private def parseSeverOptions(
+    line: CommandLine
+  ): Either[String, LanguageServerConfig] =
+    // format: off
+    for {
+      interface <- Option(line.getOptionValue(INTERFACE_OPTION))
```
Comment (kustosz) [inline engine/runner/src/main/scala/org/enso/runner/Main.scala]: Maybe worth defaulting to localhost here? That's what most of us will want 99.9% of the time, so why repeat ourselves.

File: engine/runner/src/main/scala/org/enso/runner/Main.scala [hunk 4]
```diff
@@ -164,6 +205,50 @@ object Main {
...
+                     .toRight("Interface must be provided")
+      port      <- Option(line.getOptionValue(PORT_OPTION))
```
Comment (kustosz) [inline engine/runner/src/main/scala/org/enso/runner/Main.scala]: Same here, I'd default to something like 8080, just for the sake of things being easier for testing.

Comment (lolczak) [reply engine/runner/src/main/scala/org/enso/runner/Main.scala]: fixed

File: engine/runner/src/main/scala/org/enso/runner/Main.scala (hunk 3 above)
Comment (lolczak) [reply engine/runner/src/main/scala/org/enso/runner/Main.scala]: fixed

File: engine/runner/src/main/scala/org/enso/runner/Main.scala (hunk 2 above)
Comment (lolczak) [reply engine/runner/src/main/scala/org/enso/runner/Main.scala]: fixed

File: engine/runner/src/main/scala/org/enso/runner/Main.scala (hunk 1 above)
Comment (lolczak) [reply engine/runner/src/main/scala/org/enso/runner/Main.scala]: fixed

**Pull request merged**